tests/phpunit/elementor/modules/atomic-widgets/test-atomic-widget-base.php

```diff
@@ -23,72 +25,153 @@ class Test_Atomic_Widget_Base extends Elementor_Test_Base {
+						],
+					],
+					'result' => [
+						'text' => null,
+					],
+				]
 			],
 			'transform classes' => [
-				'prop_types' => [
-					'classes' => Classes_Prop_Type::make()->default( [] ),
-					'inner_classes' => Classes_Prop_Type::make()->default( [] ),
-					'outer_classes' => Classes_Prop_Type::make()->default( [] ),
-				],
-				'settings' => [
-					'classes' => [
-						'$$type' => 'classes',
-						'value' => [ 'one', 'two', 'three' ],
+				'args' => [
+					'prop_types' => [
+						'classes' => Classes_Prop_Type::make()->default( [] ),
+						'inner_classes' => Classes_Prop_Type::make()->default( [] ),
+						'outer_classes' => Classes_Prop_Type::make()->default( [] ),
 					],
-					'outer_classes' => [
-						'$$type' => 'classes',
-						'value' => 111, // Invalid value for classes
+					'settings' => [
+						'classes' => [
+							'$$type' => 'classes',
+							'value' => [ 'one', 'two', 'three' ],
+						],
+						'outer_classes' => [
+							'$$type' => 'classes',
+							'value' => 111, // Invalid value for classes
+						],
+					],
+					'result' => [
+						'classes' => 'one two three',
+						'inner_classes' => '',
+						'outer_classes' => null,
 					],
-				],
-				'result' => [
-					'classes' => 'one two three',
-					'inner_classes' => '',
-					'outer_classes' => null,
 				],
 			],
+			'transform dynamic' => [
+				'args' => [
+					'prop_types' => [
+						'text' => String_Prop_Type::make()->default( 'Cool cool cool' ),
+						'text_2' => String_Prop_Type::make(),
+						'invalid_name' => String_Prop_Type::make(),
+						'invalid_settings' => String_Prop_Type::make(),
+					],
+					'settings' => [
+						'text' => [
+							'$$type' => 'dynamic',
+							'value' => [
+								'name' => 'dynamic-tag',
+								'settings' => [
+									'before' => 'Before text - '
+								],
+							],
+						],
+						'text_2' => [
+							'$$type' => 'dynamic',
+							'value' => [
+								'name' => 'not-exist-dynamic-tag',
+								'settings' => []
+							],
+						],
+						'invalid_name' => [
+							'$$type' => 'dynamic',
+							'value' => [
+								'name' => 123,
+								'settings' => []
+							],
+						],
+						'invalid_settings' => [
+							'$$type' => 'dynamic',
+							'value' => [
+								'name' => 'dynamic-tag',
+								'settings' => 'Invalid Before - ',
+							],
+						],
+					],
+					'result' => [
+						'text' => 'Before text - Dynamic tag content',
+						'text_2' => null,
+						'invalid_name' => null,
+						'invalid_settings' => null,
+					],
+				],
+				'arrange_cb' => function() {
+					Plugin::$instance->dynamic_tags->register(
+						new class() extends Tag {
+							public function get_name() { return 'dynamic-tag'; }
+							public function get_group() { return 'dynamic'; }
+							public function get_categories() { return [ 'text' ]; }
+							public function get_title() { return 'Dynamic Tag'; }
+							public function get_content( array $options = [] ) {
```

Comment:
I wonder if we should just create a standalone test
it's kinda complicated
WDYT?